net-misc/kea/kea-3.0.0.ebuild

```
@@ -0,0 +1,260 @@
+# Copyright 1999-2025 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{11..14} )
+inherit eapi9-ver flag-o-matic meson python-r1 systemd tmpfiles toolchain-funcs
+
+DESCRIPTION="High-performance production grade DHCPv4 & DHCPv6 server"
+HOMEPAGE="https://www.isc.org/kea/"
+
+if [[ ${PV} == 9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://gitlab.isc.org/isc-projects/kea.git"
+else
+	SRC_URI="https://downloads.isc.org/isc/kea/${PV}/${P}.tar.xz"
+	KEYWORDS="~amd64 ~arm ~arm64 ~x86"
+fi
+
+LICENSE="MPL-2.0"
+SLOT="0"
+IUSE="debug doc mysql +openssl postgres shell test"
+
+REQUIRED_USE="shell? ( ${PYTHON_REQUIRED_USE} )"
+RESTRICT="!test? ( test )"
+
+COMMON_DEPEND="
+	>=dev-libs/boost-1.66:=
+	dev-libs/log4cplus:=
+	mysql? (
+		app-arch/zstd:=
+		dev-db/mysql-connector-c:=
+		dev-libs/openssl:=
+		sys-libs/zlib:=
+	)
+	!openssl? ( dev-libs/botan:2=[boost] )
+	openssl? ( dev-libs/openssl:0= )
+	postgres? ( dev-db/postgresql:* )
+	shell? ( ${PYTHON_DEPS} )
+"
+DEPEND="${COMMON_DEPEND}
+	test? ( dev-cpp/gtest )
+"
+RDEPEND="${COMMON_DEPEND}
+	acct-group/dhcp
+	acct-user/dhcp
+"
+BDEPEND="
+	>=dev-build/meson-1.8
+	doc? (
+		$(python_gen_any_dep '
+			dev-python/sphinx[${PYTHON_USEDEP}]
+			dev-python/sphinx-rtd-theme[${PYTHON_USEDEP}]
+		')
+	)
+	virtual/pkgconfig
+"
+
+python_check_deps() {
+	use doc || return 0;
+	python_has_version "dev-python/sphinx[${PYTHON_USEDEP}]" \
+		"dev-python/sphinx-rtd-theme[${PYTHON_USEDEP}]"
+}
+
+pkg_setup() {
+	if use doc || use shell; then
+		python_setup
+	fi
+}
+
+src_prepare() {
+	default
+
+	# Remove documentation from build
+	if use !doc; then
+		sed -e "/^subdir('doc')$/d" \
+		-i meson.build || die
+	fi
+
+	# Fix up doc path where examples go
+	sed -e "s:DATADIR / 'doc/kea':DATADIR / 'doc/${P}':" \
+		-i meson.build || die
+
+	# set shebang before meson whether or not we are installing the shell
+	sed -e 's:^#!@PYTHON@:#!/usr/bin/env python3:' \
+		-i src/bin/shell/kea-shell.in || die
+
+	# Don't allow meson to install shell, we shall do that if required
+	sed -e 's:install\: true:install\: false:' \
+		-i src/bin/shell/meson.build || die
+
+	# do not create /run
+	sed -e '/^install_emptydir(RUNSTATEDIR)$/d' \
+		-i meson.build || die
+}
+
+src_configure() {
+	# https://bugs.gentoo.org/861617
+	# https://gitlab.isc.org/isc-projects/kea/-/issues/3946
+	#
+	# Kea Devs say no to LTO
+	filter-lto
+
+	local emesonargs=(
+		--localstatedir="${EPREFIX}/var"
+		-Drunstatedir="${EPREFIX}/run"
+		-Dkrb5=disabled
+		-Dnetconf=disabled
+		-Dcrypto=$(usex openssl openssl botan)
+		$(meson_feature mysql)
+		$(meson_feature postgres postgresql)
+		$(meson_feature test tests)
+	)
+	if use debug; then
+		emesonargs+=(
+			--debug
+		)
+	fi
+	meson_src_configure
+}
+
```

`doc` useflag is ineffective, it copies duplicate examples files in the wrong path (doc/kea/).
the target `doc` must be called explicity (https://gitlab.isc.org/isc-projects/kea/-/blob/Kea-3.0.0/doc/sphinx/meson.build?ref_type=tags#L147) and the path fixed.
At least, a custom src_compile phase is required:
```suggestion
src_compile() {
    meson_src_compile
    use doc && meson_src_compile doc
}
```

qa-cmp from app-portage/iwdevtools is useful to detect problems like that (files no longer installed after changes in upstream).

---

Ok that's a pain so I need sphinx as a hard build dependency whether or not doc flag is set as sphinx is used to build the man pages. By the way this is true for 2.6.X if sphinx is not install (and doc use flag not set) the man pages don't get built or installed. I guess that needs fixing? But I'll do that as a sperated PR if required.

---

Maybe it's less annoying to leave the live ebuild with manpages optional?
Anyway, the live needs some work still: [botan-3/c++20](https://gitlab.isc.org/isc-projects/kea/-/commit/88d1fb420796f4e8dd5c14cf015fd48b73d85b64)

---

>  with the hope that future tar releases will contain pre-built man pages

Could you suggest it to upstream with the function provided by eli?

---

> Anyway, the live needs some work still: [botan-3/c++20](https://gitlab.isc.org/isc-projects/kea/-/commit/88d1fb420796f4e8dd5c14cf015fd48b73d85b64)

please update the live, if devs want restore it.